### PR TITLE
Clarify use of algorithms

### LINKED ISSUE #24: resource alg properties
To address questions from Vladimir

* [ ] Remove two encryption properties `resource_encryption_alg_values_supported` `resource_encryption_enc_values_supported`
* [ ] Add explanation of `resource_signing_alg_values_supported` that it is for signing the response as described in FAPI message signing

### LINKED ISSUE #26: Update description of resource_signing_alg_values_supported
Copy language from AS metadata:

https://datatracker.ietf.org/doc/html/rfc8414#section-2

No default value if omitted, `none` must not be used.

### PR DESCRIPTION
**Pull request by selfissued**
Fixes #24 
Fixes #26 

cc @vdzhuvinov @PieterKas

### CODE REVIEW

File: draft-ietf-oauth-resource-metadata.xml [hunk 1]
```diff
@@ -243,26 +241,9 @@
 	    JSON array containing a list of the JWS <xref target="JWS" /> signing algorithms
 	    (<spanx style="verb">alg</spanx> values) <xref target="JWA" />
 	    supported by the protected resource
```
Comment (vdzhuvinov) [inline draft-ietf-oauth-resource-metadata.xml]: There should be a `.` at the end of this sentence.

**Review by vdzhuvinov — APPROVED** [review]

File: draft-ietf-oauth-resource-metadata.xml [hunk 2]
```diff
@@ -206,11 +206,9 @@
...
-	    For instance, this JWK Set MAY contain encryption key(s)
-	    that are used to encrypt access tokens to the protected resource.
-	    The JWK Set MAY also contain signing key(s) that the resource
-	    server uses to sign responses.
+	    This contains public keys belonging to the protected resource, such as
+	    signing key(s) that the resource server uses to sign resource responses.
+	    This URL MUST use the <spanx style="verb">https</spanx> scheme.
 	    When both signing and encryption keys are made available,
 	    a <spanx style="verb">use</spanx> (public key use) parameter
 	    value is REQUIRED for all keys in the referenced JWK Set
```
Comment (PieterKas) [inline draft-ietf-oauth-resource-metadata.xml:215]: @aaronpk and @selfissued, Is the intent here that if no key usage is specified, the keys are signing keys? Why not always be explicit. Either explicitly state that all keys are signing keys by default, and if encryption keys are used, key usage should be added, or alternatively make a key usage parameter mandatory, whether the keys are signing keys only, or if there are a mix of signing and encryption keys.

**Review by PieterKas — CHANGES_REQUESTED** [review]
The proposed text makes the key usage implicit, unless otherwise stated. It is also unclear if it is OK to include key usage even if there are only signing keys. 

From an implementation perspective, this creates complexity as there are now two cases to handle (1) implicit (no usage and unclear what to do if key usage is present only for signing keys - assumption is honour it, but perhaps better to be explicit) and (2) parse key usage if present. 

By removing the implicit option and always forcing key usage it removes the additional ambiguity and simplified code (i.e. always check for key usage, if no key usage, error).

Suggestion:

Replace: 

When both signing and encryption keys are made available, a <spanx style="verb">use</spanx> (public key use) parameter value is REQUIRED for all keys in the referenced JWK Set to indicate each key's intended usage.

With:

"A <spanx style="verb">use</spanx> (public key use) parameter value is REQUIRED for all keys in the referenced JWK Set to indicate each key's intended usage.

File: draft-ietf-oauth-resource-metadata.xml (hunk 2 above)
Comment (selfissued) [reply draft-ietf-oauth-resource-metadata.xml:215]: This is exactly the same language as in the `jwks_uri` definition at https://www.rfc-editor.org/rfc/rfc8414.html#section-2.  I would rather keep the OAuth metadata specs consistent in their treatment of this parameter.

Are you OK approving this on that basis, @PieterKas ?  Thanks.

Comment (PieterKas) [reply draft-ietf-oauth-resource-metadata.xml:215]: @selfissued thanks for pointing that out. 

I was thinking through whether we will break something if we move away from the implicit key usage defined in the Authorization Server Metadata spec and concluded that we probably end up making the developers life more difficult as they now have two different models for validating resource metadata, creating opportunity for confusion and reducing re-use. So assuming they are parsing and validating authorization server metadata correctly, they should be able to do the same for resource metadata. So, I think with that, I am good with approving the changes.

**Review by PieterKas — APPROVED** [review]

File: draft-ietf-oauth-resource-metadata.xml [hunk 3]
```diff
@@ -1452,6 +1436,13 @@
 	    Applied working group last call suggestions by
 	    Atul Tulshibagwale.
 	  </t>
+	  <t>
+	    Better described purpose of
+	    <spanx style="verb">resource_encryption_alg_values_supported</spanx> and
```
Comment (bc-pi) [inline draft-ietf-oauth-resource-metadata.xml]:
```suggestion
	    <spanx style="verb">resource_signing_alg_values_supported</spanx> and
```
i think is what was intended.

File: draft-ietf-oauth-resource-metadata.xml [hunk 4]
```diff
@@ -243,26 +241,10 @@
...
-
-	  <t hangText="resource_encryption_enc_values_supported">
-	    <vspace/>
-	    OPTIONAL.
-	    JSON array containing a list of the JWE encryption algorithms
-	    (<spanx style="verb">enc</spanx> values) <xref target="JWA" />
-	    supported by the protected resource
-	    for encrypted content.
+	    for signing resource responses,
+	    as described in Section 5.6.2 of <xref target="FAPI.MessageSigning"/>.
```
Comment (bc-pi) [inline draft-ietf-oauth-resource-metadata.xml]: This reads like FAPI Message Signing is the only means of signing resource responses that this is applicable to, which I don't think is intended? Maybe qualify it with a "such as described"... 

Also FWIW there is no Section 5.6.2 in the referenced version of FAPI 2.0 Message Signing.  It's in the WG/editors copy but not the published draft. 

And (draft) FAPI Message Signing uses HTTP Message Signatures, which allows for or accommodates JWS Algorithms https://datatracker.ietf.org/doc/html/rfc9421#name-json-web-signature-jws-algo but is not limited to them https://datatracker.ietf.org/doc/html/rfc9421#section-3.3 / https://www.iana.org/assignments/http-message-signature/http-message-signature.xhtml#signature-algorithms so there's kind of an impedance mismatch in this parameter that is defined as `a list of the JWS <xref target="JWS" /> signing algorithms` and applicability to FAPI Message Signing and HTTP Message Signatures in general.

File: draft-ietf-oauth-resource-metadata.xml [hunk 5]
```diff
@@ -243,26 +242,10 @@
 	    JSON array containing a list of the JWS <xref target="JWS" /> signing algorithms
 	    (<spanx style="verb">alg</spanx> values) <xref target="JWA" />
 	    supported by the protected resource
```
Comment (aaronpk) [inline draft-ietf-oauth-resource-metadata.xml]:
```suggestion
	    supported by the protected resource.
```

**Review by aaronpk — APPROVED** [review]

File: draft-ietf-oauth-resource-metadata.xml (hunk 1 above)
Comment (selfissued) [reply draft-ietf-oauth-resource-metadata.xml]: This is not the end of the sentence.  The full sentence is:
> JSON array containing a list of the JWS signing algorithms ("alg" values) supported by the protected resource for signing resource responses, for instance, as described in [FAPI.MessageSigning].

Comment (bc-pi) [reply draft-ietf-oauth-resource-metadata.xml]: now it's erroneously two sentences e3a57219cb1364185686d26ad820acece6095937

Comment (selfissued) [reply draft-ietf-oauth-resource-metadata.xml]: Corrected

**Pull request merged**